app/services/overdue_invoice_canceller.rb

```diff
@@ -0,0 +1,33 @@
+class OverdueInvoiceCanceller
+  attr_reader :invoices
+  attr_reader :delay
+
+  def initialize(invoices: Invoice.overdue, delay: self.class.delay)
+    @invoices = invoices
+    @delay = delay
+  end
+
+  def self.default_delay
+    30.days
+  end
+
+  def self.delay
```

Comment:
I guess by renaming this to `delay_in_days` we might get rid of operations on duration and treat them as integers, as does `Date`.

```ruby
def self.default_delay_in_days
  30
end

def self.delay_in_days
  Setting.days_to_keep_overdue_invoices_active || default_delay
end

Reply:
The point was to just red rid of `_days` suffix and to introduce an abstraction (a delay). It allows to keep it in any units, be it a second, minute or year.